### ISSUE TITLE
Expense - App crashes when saving expense date twice in offline mode

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#33098]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.13.0
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

**Issue found when executing PR** https://github.com/Expensify/App/pull/30737
## Action Performed:
1. Go to staging.new.expensify.com
2. Go offline
3. Navigate to the created expense details page
4. Click on the date
5. Change it once
6. Click on the date
7. Change it for the second time

## Expected Result:
App does not crash

## Actual Result:
App crashes

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/defad768-04dd-4c94-8a64-c22f87bc587b

[logs.txt](https://github.com/Expensify/App/files/13676565/logs.txt)

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by abzokhattab** [comment]
## Please re-state the problem that we are trying to solve in this issue.
Fields are blank and when changing money request date

## What is the root cause of that problem?
when changing the money request date we set onlyIncludeChangedFields param as true [here](https://github.com/Expensify/App/blob/main/src/libs/actions/IOU.js#L1016-L1020) That is why it only returns the changed
param as the money transaction
## What changes do you think we should make in order to solve the problem?
we should set it to `false` in order not to affect other fields

**Comment by bondydaa** [comment]
interesting, i guess i don't see how that is causing the crash only on the 2nd time it's called though.

cc @tgolen I see your name in that blame for `updateMoneyRequestDate` and `getUpdateMoneyRequestParams` any tips here for how to proceed?

Looks like this PR might be the root cause? https://github.com/Expensify/App/pull/30737

**Comment by paultsimura** [comment]
Here is an issue with the same RC, which was created earlier: https://github.com/Expensify/App/issues/33103
We hope to have a more fundamental discussion with Tim there 🙂

**Comment by situchan** [comment]
I think we should revert https://github.com/Expensify/App/pull/31340 (SET > MERGE) and find proper solution to fix #30290. That can be handled in #32511.

**Comment by bondydaa** [comment]
@situchan should we do anything for this GH then? Or should it be fixed by something on those PRs you've linked?

**Comment by situchan** [comment]
We should fix "crash" here. App is useless once user faced crash by updating date 2 times in offline mode.
Do you agree with my [suggestion](https://github.com/Expensify/App/issues/33098#issuecomment-1858086997)?
This will generate https://github.com/Expensify/App/issues/30290 but that should not block deploy. Better than crash

**Comment by bondydaa** [comment]
It's very unclear to me what we should do.

Sounds like on https://github.com/Expensify/App/issues/33103, https://github.com/Expensify/App/pull/31340 is also discussed that added `onyx.set` 

<img width="1032" alt="image" src="https://github.com/Expensify/App/assets/4073354/87f3e0d6-d8fe-4cca-80bc-b090b61d7ff8">

but not clear to me what the plan is there. Revert, CP fix to change `.set` to `.merge` 🤷 

If we're going to revert https://github.com/Expensify/App/pull/31340 then I think it would fix both this and https://github.com/Expensify/App/issues/33103

**Comment by situchan** [comment]
> If we're going to revert https://github.com/Expensify/App/pull/31340 then I think it would fix both this and https://github.com/Expensify/App/issues/33103

yes, correct. but the only concern is that it re-generates [this bug](https://github.com/Expensify/App/issues/30290) due to revert

**Comment by paultsimura** [comment]
@situchan https://github.com/Expensify/App/pull/31340 has just been waiting for your decision on causing a regression of https://github.com/Expensify/App/issues/32511 as well🙂 

As I described [here](https://github.com/Expensify/App/issues/30290#issuecomment-1853398810) and in the discussion of the selected proposal, using `SET` was a very risky decision. 

If we are going to revert `SET` back to `MERGE`, I would like to ask for checking [my proposal](https://github.com/Expensify/App/issues/30290#issuecomment-1777934203) to the original issue first, and then continue in https://github.com/Expensify/App/issues/32511.

**Comment by situchan** [comment]
> @situchan #31340 has just been waiting for your decision on causing a regression of #32511 as well🙂
> 
> As I described [here](https://github.com/Expensify/App/issues/30290#issuecomment-1853398810) and in the discussion of the selected proposal, using `SET` was a very risky decision.
> 
> If we are going to revert `SET` back to `MERGE`, I would like to ask for checking [my proposal](https://github.com/Expensify/App/issues/30290#issuecomment-1777934203) to the original issue first, and then continue in #32511.

Sure, it's in my next TODO. For now, focusing on this deploy blocker as requested by @trjExpensify (please assign me here as this is same as #33099 🙂)

**Comment by bondydaa** [comment]
took this to slack to discuss https://expensify.slack.com/archives/C01GTK53T8Q/p1702656464396679

**Comment by bondydaa** [comment]
okay assigning @paultsimura to spin up the revert PR of https://github.com/Expensify/App/pull/31340. 

@situchan throwing you on as the c+ then.

**Comment by jasperhuangg** [comment]
Removing Deploy Blocker label since we CP'd a revert for this

**Comment by bondydaa** [comment]
@Christinadobrzyn can you handle payment please?

@paultsimura wrote the PR https://github.com/Expensify/App/pull/33174 and @situchan was c+.

**Comment by Christinadobrzyn** [comment]
So sorry for the delay here!

Payouts due:

Issue Reporter: NA
Contributor: $500 @paultsimura (in Upwork)
Contributor+: $500 @situchan (in Upwork)

Eligible for 50% #urgency bonus? N

Upwork job is [here](https://www.upwork.com/jobs/~0122a3127ea92fcdaf).

**Comment by situchan** [comment]
We already have regression test as bug was caught during staging QA

**Comment by Christinadobrzyn** [comment]
Awesome! I've paid out this job based on this payment summary - https://github.com/Expensify/App/issues/33098#issuecomment-1875760197

I think we can close this?
